tests/functional/cli/test_cli_v4.py

```diff
@@ -252,28 +230,42 @@ def test_create_merge_request(gitlab_cli, project):
     assert ret.success
 
 
-def test_accept_request_merge(gitlab_cli, project, wait_for_sidekiq):
+def test_accept_request_merge(gitlab_cli, project):
     # MR needs at least 1 commit before we can merge
     mr = project.mergerequests.list()[0]
     file_data = {
         "branch": mr.source_branch,
-        "file_path": "README2",
+        "file_path": "test-cli-v4.md",
         "content": "Content",
-        "commit_message": "Pre-merge commit",
+        "commit_message": "chore: test-cli-v4 change",
     }
     project.files.create(file_data)
-    time.sleep(2)
-    wait_for_sidekiq(timeout=60)
-
-    cmd = [
+    # Pause to let GL catch up (happens on hosted too, sometimes takes a while for server to be ready to merge)
+    time.sleep(30)
+
+    if mr.detailed_merge_status == "not_approved":
```

Comment:
Just so we're in sync here: this doesn't check the current approved status at this point in time, but whatever the state was at the time you created the MR object when fetching the data from the server with `list()`.

But maybe that's expected, as I'm not entirely sure what the purpose of this `if` check is? Perhaps dealing with pre-existing MRs if there's some race conditions in the project tests.